app/views/user_upgrade_requests/new.html.erb

```diff
@@ -0,0 +1,20 @@
+<% set_page_title(title_with_error_prefix(t("user_upgrade_request.new.title"), @user_upgrade_request.errors.any?)) %>
+<% content_for :back_link, govuk_back_link_to(root_path, t("back_link.forms")) %>
```

Comment:
If we're going to be linking to this page from everywhere the banner is, this should probably be a plain back link?

Reply:
Any reason we can't use the back link helper in the govuk-components gem?

https://govuk-components.netlify.app/components/back-link/

Reply:
We are using the component to display the back link, it's the url needed by the component which our own application helper was intended to wrap, but I guess we have never needed.

I think above you are suggesting that the link should return to the previous page, either tasklist or forms index, which I think requires using the referer header, javascript, or setting something in the user's cookie (unless there is a better way).

That's the bit missing from our current helper and I don't think it's provided by the govuk-components gem.

Reply:
Ah yes, I see what you're talking about needing to fix in the helper now, the `back_link_url` method in `govuk_back_link_to` has no implementation 🤔 

I wonder if a compromise might be using a breadcrumb to the root page instead might be preferable?

Compromise in the sense that everyone is unhappy...

Reply:
Ah, interesting! Thanks for explaining all that.

The main scenario I can think of where using the referrer might be an issue is if a search engine indexes the page and the suggests it to someone... but that shouldn't be an issue here because the page is behind a login, right?

The dfe `back_link_url` helper has a certain neatness to it, I like how it checks that the referrer is from the same host. It's a shame that the Rails [:back URL logic](https://github.com/rails/rails/blob/fc734f28e65ef8829a1a939ee6702c1f349a1d5a/actionview/lib/action_view/helpers/url_helper.rb#L52) doesn't do the same. But it is a pain that we have to then include some extra code in all our apps we want to use it.

I think if a pentest labels this as an open redirect and we're confident that there's no issue we can push back on it.